src/btreeset.rs

```diff
@@ -567,6 +568,200 @@ where
+            }
+        }
+
+        true // No common elements
+    }
+
+    /// Returns `true` if this set is a subset of another set.
+    ///
+    /// A set `A` is a subset of a set `B` if all elements of `A` are also elements of `B`.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use ic_stable_structures::{BTreeSet, DefaultMemoryImpl};
+    ///
+    /// let mut set1: BTreeSet<u64, _> = BTreeSet::new(DefaultMemoryImpl::default());
+    /// let mut set2: BTreeSet<u64, _> = BTreeSet::new(DefaultMemoryImpl::default());
+    ///
+    /// set1.insert(1);
+    /// set1.insert(2);
+    /// set2.insert(1);
+    /// set2.insert(2);
+    /// set2.insert(3);
+    ///
+    /// assert!(set1.is_subset(&set2));
+    /// assert!(!set2.is_subset(&set1));
+    /// ```
+    pub fn is_subset(&self, other: &BTreeSet<K, M>) -> bool {
+        let mut self_iter = self.iter();
+        let mut other_iter = other.iter();
+
+        let mut self_next = self_iter.next();
+        let mut other_next = other_iter.next();
+
+        while let Some(ref self_key) = self_next {
+            match other_next {
+                Some(ref other_key) => match self_key.cmp(other_key) {
+                    std::cmp::Ordering::Equal => {
+                        // Keys match, advance both iterators.
+                        self_next = self_iter.next();
+                        other_next = other_iter.next();
+                    }
+                    std::cmp::Ordering::Greater => {
+                        // Advance the `other` iterator if its key is smaller.
+                        other_next = other_iter.next();
+                    }
+                    std::cmp::Ordering::Less => {
+                        // If `self_key` is not found in `other`, return false.
+                        return false;
+                    }
+                },
+                None => {
+                    // If `other` is exhausted but `self` is not, return false.
+                    return false;
+                }
+            }
+        }
+
+        // If we exhaust `self`, it is a subset of `other`.
+        true
+    }
+
+    /// Returns `true` if this set is a superset of another set.
+    ///
+    /// A set `A` is a superset of a set `B` if all elements of `B` are also elements of `A`.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use ic_stable_structures::{BTreeSet, DefaultMemoryImpl};
+    ///
+    /// let mut set1: BTreeSet<u64, _> = BTreeSet::new(DefaultMemoryImpl::default());
+    /// let mut set2: BTreeSet<u64, _> = BTreeSet::new(DefaultMemoryImpl::default());
+    ///
+    /// set1.insert(1);
+    /// set1.insert(2);
+    /// set1.insert(3);
+    /// set2.insert(1);
+    /// set2.insert(2);
+    ///
+    /// assert!(set1.is_superset(&set2));
+    /// assert!(!set2.is_superset(&set1));
+    /// ```
+    pub fn is_superset(&self, other: &BTreeSet<K, M>) -> bool {
```

Comment:
This does not seem particularly useful to me and I'm surprised that the Rust standard library provides it as the relationship "A is superset of B" is mathematically equivalent to "B is subset of A". Anyway if it's provided in std then it's ok for us to provide it.

Reply:
I completely agree; that was exactly what I thought.